man/systemd.exec.xml

```diff
@@ -2472,6 +2473,84 @@ StandardInputData=SWNrIHNpdHplIGRhIHVuJyBlc3NlIEtsb3BzLAp1ZmYgZWVtYWwga2xvcHAncy
+        host to services. The data is accessible from the unit's processes via
+        the file system, at a read-only location that (if possible and
+        permitted) is backed by non-swappable memory. The data is only
+        accessible to the user associated with the unit, via the
+        <varname>User=</varname>/<varname>DynamicUser=</varname> settings (as
+        well as the superuser). If enabled the location for the credential
+        objects may be retrieved via the
+        <varname>$CREDENTIALS_DIRECTORY</varname> environment variable in the
+        unit's processes.</para>
+
+        <para>The <varname>LoadCredential=</varname> setting takes a
+        textual ID to use as name for a credential plus a file system path
+        to read its data from. The ID must be a short ASCII string suitable as
+        filename in the filesystem, and may be chosen freely by the user. If the
+        specified path is absolute it is opened as regular file and the
+        credential data is read from it. If the absolute path refers to an
+        <constant>AF_UNIX</constant> stream socket in the file system a
+        connection is made to it and the credential data read from the
+        connection, providing an easy IPC integration point for dynamically
+        providing credentials from other services. If the specified path is not
+        absolute and itself qualifies as valid credential identifier it is
+        understood to refer to a credential that the service manager itself received via the
+        <varname>$CREDENTIALS_DIRECTORY</varname> environment variable, which may
+        be used to propagate credentials from an invoking environment (e.g. a
+        container manager that invoked the service manager) into a service. The credential data read from the file or
+        socket can be arbitrary binary or textual data, including newline
+        characters, NUL bytes or anything else. This option may be used multiple
+        times, each time defining an additional credential to pass to the
+        unit's processes.</para>
+
+        <para>The credential files/IPC sockets must be accessible to the
+        service manager, but don't have to be directly accessible to the unit's
+        processes: the credential data is read and copied into separate,
+        read-only copies for the unit that are accessible to appropriately
+        privileged processes. This is particular useful in combination with
+        <varname>DynamicUser=</varname> as this way privileged data can be made
+        available to processes running under a dynamic UID (i.e. a not
+        previously known one) without having to open up access to all
+        users.</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>SetCredential=</varname><replaceable>ID</replaceable>:<replaceable>VALUE</replaceable></term>
+
+        <listitem><para>The <varname>SetCredential=</varname> setting is
```

Comment:
Im not sure if we should provide this option at all. If it's public data; then a traditional path that is world-readable has the same benefits right?

Reply:
Sure, but it's nice to have this side-by-side, and I think from the application perspective it makes no sense to have different dirs to read — let's say – the public and private keys that belong together. i.e. I think philosophically it just makes sense to say we have "one dir for all creds" — a lot more than to say "please figure out the secrecy level for this kind of credential and then place it either here or there"... If you follow what I mean...

Reply:
Think about this usecase: you have some tool that checks signatures of the data i needs before processing it. It uses "signify" style sigs, i.e. short Ed25519 keys and signatures. It is just so handy to be able to parametrize this as one credential in the service file, since it is cryptographic material, but still public.